dds_web/api/user.py

```diff
@@ -68,35 +69,27 @@ def get(self):
         args = flask.request.args
         project = args.get("project")
 
-        # Check if user in db
+        # Verify username and password
         try:
-            user = models.User.query.filter(
-                models.User.username == func.binary(auth.username)
-            ).first()
+            user_ok = dds_auth.verify_user_pass(username=auth.username, password=auth.password)
         except sqlalchemy.exc.SQLAlchemyError as sqlerr:
-            return flask.make_response(f"Database connection failed: {sqlerr}", 500)
-
-        if not user:
-            return flask.make_response(f"Incorrect username and/or password!", 401)
+            return flask.make_response(str(sqlerr), 500)
+        except exceptions.AuthenticationError as autherr:
+            return flask.make_response(str(autherr), 401)
 
-        # Verify user password and generate token
-        if verify_password_argon2id(user.password, auth.password):
-            if "l" not in list(user.permissions):
-                return flask.make_response(
-                    f"The user '{auth.username}' does not have any permissions", 401
-                )
+        # User credentials incorrect
+        if not user_ok:
```

Comment:
I think this part is already covered in auth.py on lines 23-24 so expect that it will be caught on line 77.

Reply:
At the moment `verify_user_pass` raises the exception if the user doesn't exist but returns `False` if the password is incorrect. So at the moment the `if not user_ok` is needed, but I was thinking about changing

```
# Verify password and generate token
    if verify_password_argon2id(user.password, password):
        # Password correct
        return True

    # Password incorrect
    return False
```
to
```
if not verify_passwork_argon2id(user.password, password): 
    raise AuthenticationError("Incorrect username and/or password!")

return True
```
And in that case, I could remove the `if not user_ok` part -- should I do that?

Reply:
perhaps we could write it as follows

if user and verify_password_argon2id(user.password, password):
    # password correct
    return True

# if not user will come here and incorrect password as well, so we either return true or throw an exception
raise AuthenticationError("Incorrect username and/or password!")

Reply:
Sure! I actually did 
```
    # User does not exist or password does not match
    if not user or not verify_password_argon2id(user.password, password):
        raise exceptions.AuthenticationError("Incorrect username and/or password!")

    # Password correct
    return True
```

But what you wrote is better. Will switch to that.

Reply:
Will merge this though and the change will be in the next PR together with logging.